src/twisted/internet/unix.py

```diff
@@ -294,6 +294,13 @@ def getHost(self):
     def getPeer(self):
         return address.UNIXAddress(self.hostname or None)
 
+    def getTcpNoDelay(self):
```

Comment:
I think that we need a significan comment here to exlain why do we have to care about TCP things on an Unix socket

I also think that we should looks at reimplementing the inheritance for `t.i.unix.Server` so that it doesn't inherit from `tcp.Server`

It can inherit from some sort of mixin socket server that doesn't implement any interface.

```suggestion
    def getTcpNoDelay(self):
        """
        FIXME:https://github.com/twisted/twisted/issues/12369
        
        L{twisted.internet.unix.Server} inherits from L{twisted.internet.tcp.Server} which has this method implemented for TCP.
        For Unix socket, this is just a NOOP to avoid
        errors for the code that calls TCP specicific methods thinking that the Unix transport is a TCP transport.
        
```

Reply:
Added, thanks!